src/main/scala/io/qbeast/spark/internal/QbeastOptions.scala

```diff
@@ -33,13 +40,38 @@ object QbeastOptions {
     ColumnsToIndex.decode(encodedColumnsToIndex)
   }
 
+  /**
+   * Gets the desired cube size from the options
+   * @param options the options passed on the dataframe
+   * @return
+   */
+
   private def getDesiredCubeSize(options: Map[String, String]): Int = {
     options.get(CUBE_SIZE) match {
       case Some(value) => value.toInt
       case None => DEFAULT_CUBE_SIZE
     }
   }
 
+  /**
+   * Get the column stats from the options
+   * This stats should be in a JSON formatted string
+   * with the following schema
+   * {min(a):value, max(a):value, min(b):value...}
+   * @param options the options passed on the dataframe
+   * @return
+   */
+  private def getStats(options: Map[String, String]): DataFrame = {
+    val spark = SparkSession.active
+
+    options.get(STATS) match {
+      case Some(value) =>
+        import spark.implicits._
+        spark.read.json(Seq(value).toDS)
```

Comment:
Does it make sense to use spark here or is it possible to parse the options without it? (just thinking)

Reply:
Sure, probably we could think of a better approach that only uses `JSON` classes 🤔 . This is a workaround for now, that is not really "dangerous" or problematic, because those options that we are trying to parse are Spark DataFrame options (so they imply an existent `SparkSession` that we can use)